elisp/guix-env-var.el

```diff
@@ -0,0 +1,121 @@
+;;; guix-env-var.el --- Guix environment variables mode
+
+;; Copyright © 2018 Oleg Pykhalov <go.wigust@gmail.com>
+
+;; This file is part of Emacs-Guix.
+
+;; Emacs-Guix is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; Emacs-Guix is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with Emacs-Guix.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; This file provides `guix-env-var-mode' the major mode for Guix
+;; to prettify environment-variables and profile files.
+
+;;; Code:
+
+(defgroup guix-env-var nil
+  "Settings for `guix-env-var-mode'."
+  :prefix "guix-env-var-"
+  :group  'guix)
+
+(defcustom guix-env-var-enable-formatting t
+  "If non-nil, prettify Guix environment-variables and profile files."
+  :type 'boolean
+  :group 'guix-env-var)
+
+(defvar guix-env-var-display-message t
+  "If non-nil, display a message after enabling `guix-env-var-mode'.
+This message allows you to see that the current buffer is
+indented (while the original buffer file may not be indented).")
+
+(defvar guix-env-var-regexp
+  (rx "export" (char space)
+      letter (zero-or-more (or alphanumeric (syntax symbol))))
+  "Regexp matching shell variables in Guix environment-variables file.")
+
+(defvar guix-env-var-comment-line-regexp
+  (rx (zero-or-more whitespace) "#")
+  "Regexp matching a line beginning with a comment symbol.")
+
+(defvar guix-env-var-derived-sh-modes
+  '(sh-mode guix-env-var-mode)
+  "List of modes derived from `sh-mode'.")
+
+(defun guix-env-var-mode-eq (buffer mode)
+  "Return non-nil, if `major-mode' of BUFFER is equal to MODE."
+  (eq (cdr (assoc 'major-mode (buffer-local-variables buffer)))
+      mode))
+
+(defun guix-env-var-mode-p (buffer)
+  "Return t if BUFFER one of `guix-env-var-derived-sh-modes'."
+  (when (memq t (mapcar (lambda (mode)
+                          (guix-env-var-mode-eq buffer mode))
+                        guix-env-var-derived-sh-modes))
+    t))
+
+(defun guix-env-var-comment-line-p ()
+  "Return t if current line begins with a comment."
+  (save-excursion
+    (beginning-of-line)
+    (when (re-search-forward guix-env-var-comment-line-regexp
+                             (line-end-position) t)
+      t)))
```

Comment:
I would just use:

```elisp
  (re-search-forward guix-env-var-comment-line-regexp
                     (line-end-position) t)
```

without `(when ... t)` and "Return non-nil" in the docstring.